### ISSUE TITLE
Use go 1.21

### ISSUE THREAD

**Issue by szkiba** [grafana/k6foundry#5]
It would be advisable to use go 1.21 unless there is a compelling reason for 1.22.

https://github.com/grafana/k6build/blob/e0e3783398fad71aa3711fad0b996d7a14520393/go.mod#L3

**Comment by pablochacin** [comment]
@szkiba can you explain the rationale for this?

**Comment by szkiba** [comment]
Nothing special,  just if we have no reason not to follow k6 policy (support last 2 go minor version) then it seems a good idea to follow

**Comment by pablochacin** [comment]
> k6 policy (support last 2 go minor version)

k6 currently supports `1.20`

**Comment by szkiba** [comment]
Yes, I know, but I remember the last two are the policy. Of course, the last three also fulfill this.
